crates/wasi-http/src/streams_impl.rs

```diff
@@ -0,0 +1,87 @@
+use crate::poll::Pollable;
+use crate::streams::{InputStream, OutputStream, StreamError};
+use crate::WasiHttp;
+use anyhow::bail;
+use std::vec::Vec;
+
+impl crate::streams::Host for WasiHttp {
+    fn read(
+        &mut self,
+        stream: InputStream,
+        _len: u64,
```

Comment:
It doesn't seem like the `len` parameter has been used here to specify how much bytes to read. I feel like for an initial implementation this should be okay, but if the response body is too big to fit in host's memory then we probably need to use `BufRead` to read bytes into a buffer incrementally. 

I remember you added `read_into` function before, but I can't find it anymore.

Reply:
`read_into` was removed after feedback from @pchickey and others.

I fixed this function to actually implement read correctly.